lib/gauge.js

```diff
@@ -81,7 +81,11 @@ class Gauge extends Metric {
 		return startTimer.call(this, labels)();
 	}
 
-	get() {
+	async get() {
+		if (this.collect) {
+			const v = this.collect();
+			if (v instanceof Promise) await v;
```

Comment:
I think it would be cleaner to simply ensure that you can `await` things:
```js
await Promise.resolve(v)
```

Then again, NodeJS does that internally anyways, see:
```js
function foo() {
	return 2;
}

async function bar() {
	const x = foo();
	const y = await x;
	console.log(x, y); // Logs '2 2'
}

bar()
```

So the cleanest here (and below) would be to just do `await v`, no?

Reply:
The `if` avoids unnecessary ticks of the event loop by only awaiting the `collect` function if it's asynchronous. (Quite a few ticks saved even with the default metrics.)